### PR TITLE
release-2.0: storage/engine: RocksDB clear range hack

### PR DESCRIPTION
**Pull request by petermattis**
Backport 1/1 commits from #26576.

/cc @cockroachdb/release

---

Whenever we perform a ClearRange operation, also Clear the first key and
"last" key covered by the range tombstone. This allows the RocksDB
compaction code to generate better boundaries for sstables containing
range tombstones and avoids generating sstables which cover an
excessively large portion of the key space.

See #24029

Release note: None

### CODE REVIEW

**Comment by petermattis** [comment]
FYI, I'm just opening this PR so I don't forget. I'm going to hold off on merging for now.

**Review by bdarnell — APPROVED** [review]

**Comment by petermattis** [comment]
I never did merge this and now I'm not sure if we want to. This PR definitely helps with one problem with tombstones, but the more fundamental performance problems required changes to RocksDB. Do we want to backport those changes (which are quite a bit scarier)?

**Pull request closed without merging**
